web/src/app/alerts/CreateAlertDialog/StepContent/Step1.js

```diff
@@ -0,0 +1,187 @@
+      color='primary' // for white text
+      component='button'
+      label='Add All'
+      size='small'
+      icon={<AddIcon fontSize='small' />}
+      onClick={() => {
+        const toAdd = queriedServices.map(s => s.id)
+
+        // build newState
+        let newState = formFields.selectedServices
+        toAdd.forEach(s => {
+          if (newState.length < CREATE_ALERT_LIMIT) {
+            newState = newState.concat(s)
+          }
+        })
+
+        props.onChange({ selectedServices: newState })
+      }}
+    />
+  )
+
+  const serviceChips = formFields.selectedServices.map((id, key) => {
+    return (
+      <ServiceChip
+        key={key}
+        clickable={false}
+        id={id}
+        style={{ margin: 3 }}
+        onClick={e => e.preventDefault()}
+        onDelete={() =>
+          props.onChange({
+            selectedServices: formFields.selectedServices.filter(
+              sid => sid !== id,
+            ),
+          })
+        }
+      />
+    )
+  })
+
+  const notice = (
+    <Typography variant='body1' component='p' className={classes.noticeText}>
+      Select services using the search box below
+    </Typography>
+  )
+
+  return (
+    <Grid item xs={12}>
+      <InputLabel shrink>
+        {`Selected Services (${formFields.selectedServices.length})`}
+        {formFields.selectedServices.length === CREATE_ALERT_LIMIT &&
+          ' - Maximum number allowed'}
+      </InputLabel>
+      <Paper className={classes.chipContainer} elevation={0}>
+        {formFields.selectedServices.length > 0 ? serviceChips : notice}
+      </Paper>
+
+      <FormField
+        fullWidth
+        label='Search'
+        name='searchQuery'
+        fieldName='searchQuery'
+        required
+        component={TextField}
+        InputProps={{
+          ref: fieldRef,
+          startAdornment: (
+            <InputAdornment position='start'>
+              <SearchIcon color='action' />
+            </InputAdornment>
+          ),
+          endAdornment: (
+            <span className={classes.endAdornment}>
+              {queriedServices.length > 0 &&
+                formFields.selectedServices.length < CREATE_ALERT_LIMIT && (
+                  <AddAll />
+                )}
+              <ServiceLabelFilterContainer
+                value={{ labelKey, labelValue }}
+                onChange={({ labelKey, labelValue }) =>
+                  props.onChange({
+                    searchQuery: labelKey ? `${labelKey}=${labelValue}` : '',
+                  })
+                }
+                onReset={() =>
+                  props.onChange({
+                    searchQuery: '',
+                  })
+                }
+                anchorRef={fieldRef}
+              />
+            </span>
+          ),
+        }}
+      />
+
+      <List aria-label='select service options'>
+        {queriedServices.map((service, key) => (
+          <ListItem
+            button
+            key={key}
```

Comment:
Same as above, use the `service.id` instead of index so React can more efficiently update the DOM